facade/service.go

```diff
@@ -1368,21 +1411,139 @@ func (f *Facade) ScheduleServiceBatch(ctx datastore.Context, svcs []*service.Ser
 
 func (f *Facade) updateDesiredState(ctx datastore.Context, svc *service.Service, desiredState service.DesiredState) error {
 	defer ctx.Metrics().Stop(ctx.Metrics().Start("Facade.updateDesiredState"))
-	switch desiredState {
-	case service.SVCRestart:
-		// shutdown all service instances
-		if err := f.zzk.StopServiceInstances(ctx, svc.PoolID, svc.ID); err != nil {
+	logger := plog.WithFields(log.Fields{
+		"serviceid":    svc.ID,
+		"servicename":  svc.Name,
+		"desiredstate": desiredState,
+	})
+	if desiredState == service.SVCRestart && svc.DesiredState == int(service.SVCStop) {
+		// If we are already stopped or stopping, just start it back up
+		desiredState = service.SVCRun
+	}
+
+	if desiredState == service.SVCRestart {
```

Comment:
No, if the if statement above passes we still want to go into the else below (with a new desiredState of SVCRun)